### ISSUE TITLE
error: -mcpu=cortex-a7 conflicts with -march=armv7-a

### ISSUE THREAD

**Issue by james-bay** [crosstool-ng/crosstool-ng#963]
ct-ng version: 1.23.0
OS: Kbuntu 18.04 x64

I ran into a error in the build proces:
```sh
[INFO ]  Installing final gcc compiler
[EXTRA]    Configuring final gcc compiler
[EXTRA]    Building final gcc compiler
[ERROR]    cc1: error: switch -mcpu=cortex-a7 conflicts with -march=armv7-a switch [-Werror]
[ERROR]    make[6]: *** [load_1_1_.lo] Error 1
[ERROR]    make[5]: *** [all-recursive] Error 1
[ERROR]    make[4]: *** [all] Error 2
[ERROR]    make[3]: *** [all-target-libatomic] Error 2
[ERROR]    make[2]: *** [all] Error 2
```
I did not change the .config after `ct-ng armv7-rpi2-linux-gnueabihf`, I just did the following:
```sh
~/prms/crosstool-ng$ ./bootstrap
~/prms/crosstool-ng$ ./configure --enable-local
~/prms/crosstool-ng$ ./ct-ng armv7-rpi2-linux-gnueabihf
~/prms/crosstool-ng$ ./ct-ng build.1
```
- [build.log](https://cloud.mail.ru/public/B1wy/MF6nnix17)

**Comment by bhesmans** [comment]
Hey,

I ran into a similar issue, I edited the config file ... I removed `CT_ARCH_CPU="cortex-a7"` from the config file and add instead `CT_ARCH_ARCH="armv7ve"` . Now it works. I'm new to crosstool-ng and cross compile in general so it may no be a good solution at all :-P

**Comment by stilor** [comment]
This seems to be an issue with GCC8: build of libstdc++ now throws in `-march=armv7-a` unconditionally for ARM/Linux configurations - which conflicts with the `--with-cpu=...` configuration options. I asked about this on GCC mailing list, haven't got a response so far.

With setting architecture, you still need to verify how libstdc++ has been built - whether your `-march=armv7ve` came before or after `-march=armv7-a`. If your option came first, it is overridden by `-march=armv7-a`.

Until I have a better solution from GCC list, I am going to disallow GCC8 for ARM if any of these options (ARCH_ARCH, ARCH_CPU, ARCH_TUNE) is selected.

**Comment by james-bay** [comment]
@bhesmans 
@stilor 
many thanks! Большое спасибо!

**Comment by cirosantilli** [comment]
Linking to stilor's GCC thread for convenience: https://gcc.gnu.org/ml/gcc-help/2018-05/msg00120.html

Same problem with `./ct-ng arm-cortex_a15-linux-gnueabihf`.

**Comment by johan13** [comment]
This issue appears to have been fixed by #978.

**Comment by stilor** [comment]
Indeed. Forgot to mention this issue in the commit message.
